ampersand-view.js

```diff
@@ -375,8 +393,46 @@ assign(View.prototype, {
         var collectionView = new CollectionView(config);
         collectionView.render();
         return this.registerSubview(collectionView);
+    },
+
+    _setRender: function(obj) {
+        Object.defineProperty(obj, 'render', {
+            get: function() {
+                return this.__render;
+            },
+            set: function(fn) {
+                if (typeof fn !== 'function') {
+                    throw new ReferenceError('render() must be a function. ' +
+                        'received: ' + fn.toString());
+                }
+                this.__render = function() {
+                    fn.apply(this, arguments);
+                    return this._render();
+                };
+            }
+        });
+    },
+
+    _setRemove: function(obj) {
+        Object.defineProperty(obj, 'remove', {
+            get: function() {
+                return this.__remove;
+            },
+            set: function(fn) {
+                if (typeof fn !== 'function') {
```

Comment:
lol, my comment made no sense, sorry. A better example is we don't check to see if you overwrite `initialize` with a function or something else.